Feeder audit 2026-05-29: usgs-nwis-wq

## Feeder audit summary for `usgs-nwis-wq`

This sweep covers the objective repo-state parts of the feeder-release-checklist: required file presence, catalog + ghpages source entries, source-local pytest with generated producer src paths injected into PYTHONPATH, Docker E2E matrix coverage, ARM static audit, Fabric static audit, and ARM generator dry-run no-diff. It does not rerun live Azure/Fabric deploy bullets, specialist-agent reviews, or upstream API re-audits.

## Findings

- **ERROR** `missing-docker-build-matrix-entry` — 1 finding(s)
  - tests/docker_e2e/matrix.json has no build entry for kafka (Dockerfile). (tests/docker_e2e/matrix.json)
- **ERROR** `missing-docker-flow-matrix-entry` — 1 finding(s)
  - tests/docker_e2e/matrix.json has no flow entry for kafka (Dockerfile). (tests/docker_e2e/matrix.json)
- **ERROR** `missing-file` — 2 finding(s)
  - Required file missing: feeders/usgs-nwis-wq/xreg/usgs-nwis-wq.xreg.json (feeders/usgs-nwis-wq/xreg/usgs-nwis-wq.xreg.json)
  - Required file missing: feeders/usgs-nwis-wq/kql/usgs-nwis-wq.kql (feeders/usgs-nwis-wq/kql/usgs-nwis-wq.kql)
- **ERROR** `pytest-failed` — 1 finding(s)
  - Source-local pytest failed. (feeders/usgs-nwis-wq/tests)
- **WARNING** `arm-stub-description` — 2 finding(s)
  - azure-template-with-eventhub.json: parameter 'eventHubSku' has stub or missing metadata.description (feeders/usgs-nwis-wq/azure-template-with-eventhub.json)
  - azure-template-with-servicebus.json: parameter 'queueName' has stub or missing metadata.description (feeders/usgs-nwis-wq/azure-template-with-servicebus.json)

## Check evidence

### `arm_audit` — rc=0, 1.31s

```text
ARM template static audit
  RepoRoot:        C:\git\real-time-sources
  Feeders scanned: 1
  Templates:       5
  Blockers:        0
  Warnings:        2

[usgs-nwis-wq] blockers=0 warnings=2
  azure-template-with-eventhub.json
    [WARNING] stub-description: parameter 'eventHubSku' has stub or missing metadata.description
  azure-template-with-servicebus.json
    [WARNING] stub-description: parameter 'queueName' has stub or missing metadata.description
PASS (with 2 warning(s)).
Report written to C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\arm\usgs-nwis-wq.json
```

### `arm_generator` — rc=0, 3.13s

```text
Description sources: total=177, container=31, readme=0, argparse=28, fallback=118
Generated 0 ARM template(s).
```

### `fabric_audit` — rc=0, 1.25s

```text
[32;1mSource      [0m[32;1m OptIn[0m[32;1m Notebook[0m[32;1m  Hook[0m[32;1m PASS[0m[32;1m WARN[0m[32;1m BLOCK[0m
[32;1m------      [0m [32;1m-----[0m [32;1m--------[0m [32;1m ----[0m [32;1m----[0m [32;1m----[0m [32;1m-----[0m
usgs-nwis-wq  True     True False   10    0     0

Sources: 1 total, 0 blockers, 0 warnings
JSON report: C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\fabric\usgs-nwis-wq.json
```

### `pytest` — rc=1, 3.64s

```text
============================= test session starts =============================
platform win32 -- Python 3.11.9, pytest-8.4.2, pluggy-1.6.0
rootdir: C:\git\real-time-sources\feeders\usgs-nwis-wq
configfile: pytest.ini
plugins: anyio-4.9.0, asyncio-1.2.0, cov-6.0.0, timeout-2.4.0, requests-mock-1.12.1
asyncio: mode=Mode.STRICT, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 0 items / 1 error

=================================== ERRORS ====================================
__________________ ERROR collecting tests/test_once_mode.py ___________________
ImportError while importing test module 'C:\git\real-time-sources\feeders\usgs-nwis-wq\tests\test_once_mode.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2544.0_x64__qbz5n2kfra8p0\Lib\importlib\__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests\test_once_mode.py:8: in <module>
    from usgs_nwis_wq.usgs_nwis_wq import USGSWaterQualityPoller
usgs_nwis_wq\usgs_nwis_wq.py:26: in <module>
    from usgs_nwis_wq_producer_data.usgs_nwis_wq_producer_data.monitoringsite import MonitoringSite
E   ModuleNotFoundError: No module named 'usgs_nwis_wq_producer_data.usgs_nwis_wq_producer_data'
=========================== short test summary info ===========================
ERROR tests/test_once_mode.py
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
============================== 1 error in 0.84s ===============================
```

## Next step

Fix the findings above, then rerun the sequential feeder-release-checklist audit for this feeder and close this issue with the new evidence.

--- a/feeders/usgs-nwis-wq/usgs_nwis_wq/usgs_nwis_wq.py
+++ b/feeders/usgs-nwis-wq/usgs_nwis_wq/usgs_nwis_wq.py
@@ -23,8 +23,8 @@ from typing import Dict, List, Optional, Tuple, Any, Set
 from confluent_kafka import Producer
 
 # pylint: disable=import-error, line-too-long
-from usgs_nwis_wq_producer_data.usgs_nwis_wq_producer_data.monitoringsite import MonitoringSite
-from usgs_nwis_wq_producer_data.usgs_nwis_wq_producer_data.waterqualityreading import WaterQualityReading
+from usgs_nwis_wq_producer_data.monitoringsite import MonitoringSite
+from usgs_nwis_wq_producer_data.waterqualityreading import WaterQualityReading
 from usgs_nwis_wq_producer_kafka_producer.producer import (
     USGSWaterQualitySitesEventProducer,
     USGSWaterQualityReadingsEventProducer,
@@ -164,6 +164,13 @@ class USGSWaterQualityPoller:
             latitude = geo.get("latitude")
             longitude = geo.get("longitude")
 
+            # Extract variable info
+            var_codes = variable.get("variableCode", [])
+            parameter_code = var_codes[0].get("value", "") if var_codes else ""
+            parameter_name = html.unescape(variable.get("variableDescription", variable.get("variableName", "")))
+            unit_code = variable.get("unit", {}).get("unitCode", "")
+            no_data_value = variable.get("noDataValue", USGSWaterQualityPoller.NO_DATA_VALUE)
+
             # Extract site properties
             site_type = None
             state_code = None
@@ -194,15 +201,10 @@ class USGSWaterQualityPoller:
                     state_code=state_code,
                     county_code=county_code,
                     huc_code=huc_code,
+                    state=None,
+                    parameter_code=parameter_code or None,
                 )
                 sites_list.append(site)
-
-            # Extract variable info
-            var_codes = variable.get("variableCode", [])
-            parameter_code = var_codes[0].get("value", "") if var_codes else ""
-            parameter_name = html.unescape(variable.get("variableDescription", variable.get("variableName", "")))
-            unit_code = variable.get("unit", {}).get("unitCode", "")
-            no_data_value = variable.get("noDataValue", USGSWaterQualityPoller.NO_DATA_VALUE)
 
             # Extract timezone info
             tz_info = source_info.get("timeZoneInfo", {})
@@ -253,6 +255,7 @@ class USGSWaterQualityPoller:
                         unit=unit_code,
                         qualifier=qualifier,
                         date_time=dt_utc.isoformat(),
+                        state=None,
                     )
                     readings_list.append(reading)
 

